### PR TITLE
Prevent clashing schema names

### PR DESCRIPTION
**Pull request by improbable-valentyn**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Sometimes class names will clash in schema as a result of classes with the same name having a different asset path or the schema name ending up the same after removing non-alphanumeric characters. This will prevent clashing schema names by appending a number to the resulting name until there's no clash.

It is a rare case that the dev shouldn't aim for anyway, so I went for this simpler approach as opposed to e.g. embedding the class path in the schema component names.

Note that this could generate confusing results in edge cases. E.g.:
`/Game/Box`, `/Game/Debug/Box`, and `/Game/Box1` could end up generating different schema components depending on the order.
`/Game/Box` -> `/Game/Debug/Box` -> `/Game/Box1` will give `Box` -> `Box1` -> `Box11`, but
`/Game/Box` -> `/Game/Box1` -> `/Game/Debug/Box` will give `Box` -> `Box1` -> `Box2`
#### Tests
- Have two classes with the same name but different path
- Generate schema successfully
#### Primary reviewers
@m-samiec @Vatyx

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/Utils/DataTypeUtilities.cpp [hunk 1]
```diff
@@ -5,11 +5,11 @@
 #include "Algo/Transform.h"
 
 // Regex pattern matcher to match alphanumeric characters.
-const FRegexPattern AlphanumericPattern(TEXT("[A-Z,a-z,0-9]"));
+const FRegexPattern AlphanumericPattern(TEXT("[A-Za-z0-9]"));
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/Utils/DataTypeUtilities.cpp]: Drive-by fix since the previous regex actually included the character `,`.

File: SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp [hunk 2]
```diff
@@ -122,23 +126,24 @@ bool CheckIdentifierNameValidity(TSharedPtr<FUnrealType> TypeInfo)
 bool ValidateIdentifierNames(TArray<TSharedPtr<FUnrealType>>& TypeInfos)
 {
 	// Remove all underscores from the class names, check for duplicates.
-	for (int i = 0; i < TypeInfos.Num() - 1; ++i)
+	for (auto& TypeInfo : TypeInfos)
```
Comment (Vatyx) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp]:
```suggestion
	for (const auto& TypeInfo : TypeInfos)
```
